### PR TITLE
nodeup: default nonMasqueradeCIDR to the podCIDR

### PR DESCRIPTION
**Pull request by justinsb**
Rather than failing, we can at least try the podCIDR here, as it is
likely the value we want.

### CODE REVIEW

File: nodeup/pkg/model/containerd.go [hunk 1]
```diff
@@ -407,12 +407,17 @@ iptables -w -t nat -A IP-MASQ -d {{.NonMasqueradeCIDR}} -m comment --comment "ip
...
 `
 
-	if b.Cluster.Spec.Networking.NonMasqueradeCIDR == "" {
-		// We could fall back to the pod CIDR, that is likely more correct anyway
-		return fmt.Errorf("NonMasqueradeCIDR is not set")
+	nonMasqueradeCIDR := b.Cluster.Spec.Networking.NonMasqueradeCIDR
+	if nonMasqueradeCIDR == "" {
+		if podCIDR := b.Cluster.Spec.Networking.PodCIDR; podCIDR != "" {
+			klog.Warningf("defaulting nonMasqueradeCIDR to podCIDR %q", podCIDR)
+			nonMasqueradeCIDR = podCIDR
```
Comment (johngmyers) [inline nodeup/pkg/model/containerd.go:414]: It's going to masquerade the `ServiceClusterIPRange`? That doesn't seem correct.

File: nodeup/pkg/model/containerd.go [hunk 2]
```diff
@@ -407,12 +407,17 @@ iptables -w -t nat -A IP-MASQ -d {{.NonMasqueradeCIDR}} -m comment --comment "ip
 iptables -w -t nat -A IP-MASQ -m comment --comment "ip-masq: outbound traffic is subject to MASQUERADE (must be last in chain)" -j MASQUERADE
 `
 
-	if b.Cluster.Spec.Networking.NonMasqueradeCIDR == "" {
-		// We could fall back to the pod CIDR, that is likely more correct anyway
-		return fmt.Errorf("NonMasqueradeCIDR is not set")
+	nonMasqueradeCIDR := b.Cluster.Spec.Networking.NonMasqueradeCIDR
+	if nonMasqueradeCIDR == "" {
+		if podCIDR := b.Cluster.Spec.Networking.PodCIDR; podCIDR != "" {
+			klog.Warningf("defaulting nonMasqueradeCIDR to podCIDR %q", podCIDR)
```
Comment (johngmyers) [inline nodeup/pkg/model/containerd.go:413]: Why not populate `NonMasqueradeCIDR` in the completed cluster spec, perhaps in `performNetworkAssignmentsIPAliases()`?

If iptables is disabling masquerading for the CIDR, then it truly is the `NonMasqueradeCIDR`. Why should GCP in particular bind the value so much later than other cloud providers?

Comment (olemarkus) [reply nodeup/pkg/model/containerd.go:413]: I would also do this in cloudup. Doing it here would be a bit surprising and I would probably not think to look here.

File: nodeup/pkg/model/containerd.go (hunk 1 above)
Comment (justinsb) [reply nodeup/pkg/model/containerd.go:414]: The service ips are dealt with "earlier" ... if we put a service IP onto the wire (i.e. dev eth0), then - yes - we would NAT it.

**Pull request closed without merging**